temporal-sdk/src/main/java/io/temporal/internal/common/GrpcRetryer.java

```diff
@@ -205,9 +209,14 @@ public Throwable getException() {
     if (e == null) {
       return new ValueExceptionPair<>(CompletableFuture.completedFuture(r), null);
     }
+    if (e instanceof CompletionException) {
+      e = e.getCause();
```

Comment:
This is the fix

Reply:
Can you leave a comment explaining why we do what we do?
Also it's possible for CompletionException to be constructed without cause, should we still do this trick in that case?

Reply:
> Can you leave a comment explaining why we do what we do?
I added a comment, plus now there are tests for both wrapped and unwrapped exceptions.

> Also it's possible for CompletionException to be constructed without cause, should we still do this trick in that case?
It's a kinda illegal state. CompletionException is intended to be a wrapper over another exception. But it's technically possible and it will work correctly: The next `if` check will be true and we will return without retrying because GrpcRetrier retries only StatusRuntimeException.